app/src/features/Snackbar/snackbarSlice.ts

```diff
@@ -0,0 +1,52 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+import { SnackbarKey, VariantType } from "notistack";
+
+export type SnackbarNotificationState = {
+  message: string;
+  key: SnackbarKey;
+  options: { key: SnackbarKey; variant: VariantType };
+};
+
+type SnackbarSliceState = {
+  notifications: SnackbarNotificationState[];
+};
+
+const initialState: SnackbarSliceState = {
+  notifications: [],
+};
+
+const snackbarSlice = createSlice({
+  name: "snackbar",
+  initialState,
+  reducers: {
```

Comment:
Maybe you should use `createEntityAdapter` from RTK, it's perfectly adapted for the kind of actions you're implementing

Have a look to other slices to get an exemple ;)